File: python/ex2.py
from enum import Enum
from typing import List


class NecklaceType(Enum):
    Uniq = 1
    Boring = 2
    Regular = 3
    Perfect = 4


class Necklace:
    green: int
    red: int
    yellow: int

    def __init__(self, green: int, red: int, yellow: int) -> None:
        self.green = green
        self.red = red
        self.yellow = yellow

    def getNecklaceType(self) -> NecklaceType:
        if (self.green != 0 and self.yellow == 0 and self.red == 0 or self.green == 0 and self.yellow == 0 and self.red != 0
                or self.green == 0 and self.yellow != 0 and self.red == 0):
            return NecklaceType.Uniq
        elif (self.green == self.yellow and self.yellow == self.red):
            return NecklaceType.Perfect
        elif (self.green != 0 and self.yellow != 0 and self.red == 0
              or self.green == 0 and self.yellow != 0 and self.red != 0
              or self.green != 0 and self.yellow == 0 and self.red != 0):
            return NecklaceType.Boring
        else:
            return NecklaceType.Regular

    def __len__(self):
        return self.green+self.red+self.yellow


def how_many_boring_necklaces(necklaces: List[Necklace], num: int) -> int:
    counter = 0
    for nck in necklaces:
        if (nck.getLength() > num and nck.getNecklaceType() == NecklaceType.Boring):
            counter += 1
    return counter


def main():
    pass


if __name__ == '__main__':
    main()

File: Java/src/Ex2.java
enum NecklaceType {
    Uniq, Boring, Regular, Perfect
}

class Necklace {
    int green, yellow, red;

    Necklace(int green, int yellow, int red) {
        this.green = green;
        this.yellow = yellow;
        this.red = red;
    }

    NecklaceType getNecklaceType() {
        if (this.green != 0 && this.yellow == 0 && this.red == 0 || this.green == 0 && this.yellow == 0 && this.red != 0
                || this.green == 0 && this.yellow != 0 && this.red == 0) {
            return NecklaceType.Uniq;
        } else if (this.green == this.yellow && this.yellow == this.red) {
            return NecklaceType.Perfect;
        } else if (this.green != 0 && this.yellow != 0 && this.red == 0
                || this.green == 0 && this.yellow != 0 && this.red != 0
                || this.green != 0 && this.yellow == 0 && this.red != 0) {
            return NecklaceType.Boring;
        } else {
            return NecklaceType.Regular;
        }
    }

    int getLength() {
        return this.green + this.yellow + this.red;
    }
}

public class Ex2 {
    int howManyBoringNecklaces(Necklace[] necklaces, int num) {
        int counter = 0;
        for (Necklace nck : necklaces) {
            if (nck.getLength() > num && nck.getNecklaceType() == NecklaceType.Boring) {
                counter++;
            }
        }
        return counter;
    }

    public static void main(String[] args) {

    }
}
